cartographer/mapping/internal/2d/local_trajectory_builder_2d.cc

```diff
@@ -259,13 +262,11 @@ LocalTrajectoryBuilder2D::InsertIntoSubmap(
     return nullptr;
   }
 
-  // Querying the active submaps must be done here before calling
-  // InsertRangeData() since the queried values are valid for next insertion.
+  active_submaps_.InsertRangeData(range_data_in_local);
   std::vector<std::shared_ptr<const Submap2D>> insertion_submaps;
   for (const std::shared_ptr<Submap2D>& submap : active_submaps_.submaps()) {
```

Comment:
Can we change `submaps()` to return a ` std::vector<std::shared_ptr<const Submap2D>>`? Similar in 3D.

Reply:
Do you want to move the for-loop into the getter?
There is no `submaps()` getter in 3D that returns `std::vector<std::shared_ptr<const Submap2D>>`.

Reply:
Okay, I will re-construct the vector with `const Submap2D`.